pkg/scan/scan.go

```diff
@@ -329,3 +347,160 @@ func getInterfaceFromIP(ip net.IP) (*net.Interface, error) {
 
 	return nil, fmt.Errorf("no interface found for ip %s", address)
 }
+
+// ConnectPort a single host and port
+func ConnectPort(host string, port int, timeout time.Duration) (bool, error) {
+	conn, err := net.DialTimeout("tcp", fmt.Sprintf("%s:%d", host, port), timeout)
+	if err != nil {
+		return false, err
+	}
+	defer conn.Close()
+	return true, err
+}
+
+func (s *Scanner) ACKPort(dstIP string, port int, timeout time.Duration) (bool, error) {
+	conn, err := net.ListenPacket("ip4:tcp", "0.0.0.0")
+	if err != nil {
+		return false, err
+	}
+	defer conn.Close()
+
+	rawPort, err := freeport.GetFreePort()
+	if err != nil {
+		return false, err
+	}
+
+	// Construct all the network layers we need.
+	ip4 := layers.IPv4{
+		SrcIP:    s.SourceIP,
+		DstIP:    net.ParseIP(dstIP),
+		Version:  4,
+		TTL:      255,
+		Protocol: layers.IPProtocolTCP,
+	}
+	tcpOption := layers.TCPOption{
+		OptionType:   layers.TCPOptionKindMSS,
+		OptionLength: 4,
+		OptionData:   []byte{0x12, 0x34},
+	}
+	randSeq := 1000000000 + rand.Intn(math.MaxInt32)
```

Comment:
Instead of opting for a random sequence, let's use an incremental sequence increased on each run since i've noticed many targets dropping masscan/other scanners probes because they use random sequence. Hence, this should be changed too.

Reply:
Done